elastic/kibana
# [Synthetics] Apply faster backoff for non-`test-now` screenshot results

## Summary

Resolves #152754.

It was noted in #152754 that we're getting some negative side effects from the exponential backoff we added to the getter for journey screenshots. When a screenshot is legitimately not available for an older journey (as opposed to not yet fully indexed for a fresh journey), the backoff will cause the screenshot never to render.

## Testing

Follow a flow similar to what's illustrated in the GIF below.

1. Create a monitor with some screenshots that show up.
2. Delete the screenshot data stream. A good way is to use the Index Management > Data Streams view in Kibana.
3. Refresh your monitor's detail page when new photos are coming in. It should give up on finding the images very fast, and as they aren't there, you should see the missing image placeholder.
4. Start creating a new monitor. Make it have multiple steps.
5. Ensure that you aren't seeing any "infinite" loading states across the screenshot UIs.

![20230308110103](https://user-images.githubusercontent.com/18429259/223766486-eaca4295-9e4f-4490-8717-c1b915382ec7.gif)


### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers
